framework/src/base/MooseApp.C

```diff
@@ -247,6 +247,10 @@ MooseApp::validParams()
       "Continue the calculation. Without <file base>, the most recent recovery file will be used");
   params.setGlobalCommandLineParam("recover");
 
+  params.addCommandLineParam<bool>(
+      "suppress_header", "--suppress-header", "Flag to print the App header");
+  params.setGlobalCommandLineParam("suppress_header");
```

Comment:
Ugh, this comment got stuck in "review", sorry it didn't go out...anyways, heres the command line and old comment...

@loganharbour This PR builds on #29663 , so it's a bit convoluted with that update, but here is the cmd line I'm trying to add. I would like to be able to optionally suppress the header being printed. I tried to use `CommonOutputAction`, similar to #29665, but it's not built at this point (I think?).